.ddev/.typo3-setup/scripts/.utils.sh

```
@@ -0,0 +1,311 @@
+    composer config extra.typo3/cms.web-dir public --working-dir "$BASE_PATH"
+    composer config repositories.packages path 'packages/*' --working-dir "$BASE_PATH"
+    composer config --no-interaction allow-plugins.typo3/cms-composer-installers true --working-dir "$BASE_PATH"
+    composer config --no-interaction allow-plugins.typo3/class-alias-loader true --working-dir "$BASE_PATH"
+}
+
+# Function to set up TYPO3 configuration.
+# It changes the current directory to the base path, sets the TYPO3 installation database name,
+# and configures various TYPO3 settings such as debug mode, error display, trusted hosts pattern,
+# mail transport, and graphics processor.
+function setup_typo3() {
+    cd $BASE_PATH
```

Shellcheck (0.10.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)